src/api/RequestApi.ts

```diff
@@ -0,0 +1,198 @@
+import { spWebContext } from "../providers/SPWebContext";
+import { ApiError } from "./InternalErrors";
+import { IItemUpdateResult } from "@pnp/sp/items";
+import { emptype } from "../constants/EmpTypes";
+import { worklocation } from "../constants/WorkLocations";
+import { SPPersona } from "../components/PeoplePicker/PeoplePicker";
+
+// create PnP JS response interface for the InForm
+// This extends the IInForm -- currently identical, but may need to vary when pulling in SPData
+type IResponseItem = IInForm;
+
+// create IItem item to work with it internally
+export type IInForm = {
+  /** Required - Will be -1 for NewForms that haven't been saved yet */
+  Id: number;
+  /** Required - Contains the Employee's Name */
+  empName: string;
+  /** Required - Employee's Type valid values are:
+   * 'civ' - for Civilian Employees
+   * 'mil' - for Military Employees
+   * 'ctr' - for Contracted Employees
+   */
+  empType: emptype;
+  /** Required - The Employee's Grade/Rank.  Not applicable if 'ctr' */
+  gradeRank: string;
+  /** Required - Possible values are 'local' and 'remote'  */
+  workLocation: worklocation;
+  /** Required - The Employee's Office */
+  office: string;
+  /** Required - Can only be 'true' if it is a New to USAF Civilain.  Must be 'false' if it is a 'mil' or 'ctr' */
+  // TODO - Look into making this a Type or Leveraging SharePoint Type -- Think they possibly use yes/no instead of true/false
+  isNewCiv: "yes" | "no";
+  /** Required - The user's previous organization.  Will be "" if isNewCiv is false */
+  prevOrg: string;
+  /** Required - The user's Estimated Arrival Date */
+  eta: Date;
+  /** Required - The Superviosr/Gov Lead of the employee */
+  supGovLead: SPPersona[];
+};
+
+/** For new forms, allow certain fields to be blank or undefined to support controlled components */
+export type INewInForm = Omit<
+  IInForm,
+  "empType" | "workLocation" | "eta" | "supGovLead" | "isNewCiv"
+> & {
+  empType: emptype | "";
+  workLocation: worklocation | "";
+  eta: Date | undefined;
+  supGovLead: SPPersona[] | undefined;
+  isNewCiv: "yes" | "no" | "";
+};
+
+export interface IInFormApi {
+  /**
+   * Gets the request based on ID.
+   *
+   * @param ID The ID of the item to retrieve from SharePoint
+   * @returns The IITem for the given ID
+   */
+  getItemById(ID: number): Promise<IResponseItem | undefined>;
+
+  /**
+   * Update/persist the given Item
+   *
+   * @param requirementsRequest The RequirementsRequest to be saved/updated
+   */
+  updateItem(IItem: IInForm): Promise<IItemUpdateResult>;
+}
+
+export class RequestApi implements IInFormApi {
+  itemList = spWebContext.web.lists.getByTitle("Items");
+
+  async getItemById(ID: number): Promise<IResponseItem | undefined> {
+    try {
+      // use map to convert IResponseItem[] into our internal object IItem[]
+      const response: IResponseItem = await this.itemList.items.getById(ID)();
```

Comment:
I think we can type these better without having to do all the manual mapping
`const response: IResponseItem = await this.itemList.items.getById(ID)<IInform>();`
or I guess
`const response: IResponseItem = await this.itemList.items.getById(ID)<IResponseItem>();`
then we can just
`return response;`

Same for the updateItem function.

Reply:
I was wondering about this, but I think PnPJS doesn't actaully do very good typing, so we have to map them ourselves.  Unless of course we leverage the select qualifier, then looks like you can specify a type coming back...  Not sure how this works with using luxon, people, etc.
https://pnp.github.io/pnpjs/concepts/typings/